clients/geth/specular/rollup/services/sequencer/sequencer.go

```diff
@@ -59,28 +60,131 @@ func New(eth services.Backend, proofBackend proof.Backend, cfg *services.Config,
 	defer txsSub.Unsubscribe()
 
+	// Process txns via batcher
 	batcher, err := NewBatcher(s.Config.Coinbase, s.Eth)
 	if err != nil {
 		log.Crit("Failed to start batcher", "err", err)
 	}
 
+	// Loop over txns
 	for {
 		select {
+		case <-timer.C:
+			// Get pending txs - locals and remotes, sorted by price
+			var txs []*types.Transaction
+			var sortedTxs *types.TransactionsByPriceAndNonce
+			signer := types.MakeSigner(batcher.chainConfig, batcher.header.Number)
+			pending := s.Eth.TxPool().Pending(true)
+			localTxs, remoteTxs := make(map[common.Address]types.Transactions), pending
+			for _, account := range s.Eth.TxPool().Locals() {
+				if txs = remoteTxs[account]; len(txs) > 0 {
+					delete(remoteTxs, account)
+					localTxs[account] = txs
+				}
+			}
+			if len(localTxs) > 0 {
+				sortedTxs = types.NewTransactionsByPriceAndNonce(signer, localTxs, batcher.header.BaseFee)
+			}
+			if len(remoteTxs) > 0 {
+				sortedTxs = types.NewTransactionsByPriceAndNonce(signer, remoteTxs, batcher.header.BaseFee)
+			}
+
+			// Batch txns
+			if sortedTxs == nil {
+				timer.Reset(timeInterval)
+				break
+			}
+			for {
+				tx := sortedTxs.Peek()
+				if tx == nil {
+					break
+				}
+				prevTx, _, _, _, err := s.ProofBackend.GetTransaction(s.Ctx, tx.Hash())
+				if err != nil {
+					sortedTxs.Pop()
+					continue
+				}
+				if prevTx != nil {
+					txs = removeTx(txs, prevTx)
+				} else {
+					txs = append(txs, tx)
+				}
+				sortedTxs.Pop()
+			}
+
+			// filter txs
+			for i := len(txs) - 1; i >= 0; i-- {
+				tx := txs[i]
+				prevTx, _, _, _, err := s.ProofBackend.GetTransaction(s.Ctx, tx.Hash())
+				if err != nil {
+					log.Error("Filtering via GetTransaction(), this is err", "err", err)
+					continue
+				}
+				if prevTx != nil {
+					txs = removeTx(txs, prevTx)
+				}
+			}
+
+			if len(txs) == 0 {
+				timer.Reset(timeInterval)
+				break
+			}
+			err = batcher.CommitTransactions(txs)
+			if err != nil {
+				log.Crit("Failed to commit transactions", "err", err)
+			}
+			blocks, err := batcher.Batch()
+			if err != nil {
+				log.Crit("Failed to batch blocks", "err", err)
+			}
+			batch := rollupTypes.NewTxBatch(blocks, 0) // TODO: add max batch size
+			s.batchCh <- batch
+			timer.Reset(timeInterval)
 		case ev := <-txsCh:
-			// New transactions from TxPool
-			err = batcher.CommitTransactions(ev.Txs)
+			// Batch txs in case of txEvent
+			var batchTxs []*types.Transaction
+			txs := make(map[common.Address]types.Transactions)
+			signer := types.MakeSigner(batcher.chainConfig, batcher.header.Number)
+			for _, tx := range ev.Txs {
+				acc, _ := types.Sender(signer, tx)
+				txs[acc] = append(txs[acc], tx)
+			}
+			sortedTxs := types.NewTransactionsByPriceAndNonce(signer, txs, batcher.header.BaseFee)
+			for {
+				tx := sortedTxs.Peek()
+				if tx == nil {
+					break
+				}
+				batchTxs = append(batchTxs, tx)
+				sortedTxs.Pop()
+			}
+			err = batcher.CommitTransactions(batchTxs)
```

Comment:
if you commit transactions here, the ticker would be useless. instead you should cache all the received transactions here in a local var like sortedTxs, then when ticker ticks you want to 1. fetch all pending tx from txpool; 2. insert them into the sortedTxs var if (a) tx not in sortedTxs (b) tx not in blockchain; 3. commit sortedTxs and clear it